aws-cpp-sdk-core/source/utils/DateTime.cpp

```diff
@@ -802,6 +802,21 @@ class ISO_8601DateParser : public DateParser
                     }
 
                     break;
+                case 6:
+                    if (c == 'Z' && index - stateStartIndex == 3)
+                    {
+                        m_state = finalState;
+                    }
+                    else if (isdigit(c))
+                    {
+						//skip
+                    }
+                    else
+                    {
+                        m_error = true;
+                    }
+
+                    break;
```

Comment:
This last state needs to be optional. Most services do not send the MS precision, however it they do, we need to parse correctly. I'll merge these two this morning and push it out.

Thanks for your work on this and for pointing out this edge case.